components/tools/OmeroWeb/test/integration/test_api_images.py

```diff
@@ -0,0 +1,182 @@
+    return user[0].getSession().getQueryService()
+
+
+def get_connection(user, group_id=None):
+    """Get a BlitzGateway connection for the given user's client."""
+    connection = BlitzGateway(client_obj=user[0])
+    # Refresh the session context
+    connection.getEventContext()
+    if group_id is not None:
+        connection.SERVICE_OPTS.setOmeroGroup(group_id)
+    return connection
+
+
+def cmp_name_insensitive(x, y):
+    """Case-insensitive name comparator."""
+    return cmp(unwrap(x.name).lower(), unwrap(y.name).lower())
+
+
+def marshal_objects(objects):
+    """Marshal objects using omero_marshal."""
+    expected = []
+    for obj in objects:
+        encoder = get_encoder(obj.__class__)
+        expected.append(encoder.encode(obj))
+    return expected
+
+
+def assert_objects(conn, json_objects, omero_ids_objects, dtype="Project",
+                   group='-1', extra=None, opts=None):
+    """
+    Load objects from OMERO, via conn.getObjects().
+
+    marshal with omero_marshal and compare with json_objects.
+    omero_ids_objects can be IDs or list of omero.model objects.
+
+    @param: extra       List of dicts containing expected extra json data
+                        e.g. {'omero:childCount': 1}
+    """
+    pids = []
+    for p in omero_ids_objects:
+        try:
+            pids.append(long(p))
+        except TypeError:
+            pids.append(p.id.val)
+    conn.SERVICE_OPTS.setOmeroGroup(group)
+    objs = conn.getObjects(dtype, pids, respect_order=True, opts=opts)
+    objs = [p._obj for p in objs]
+    expected = marshal_objects(objs)
+    assert len(json_objects) == len(expected)
+    for i, o1, o2 in zip(range(len(expected)), json_objects, expected):
+        if extra is not None and i < len(extra):
+            o2.update(extra[i])
+        # dumping to json and loading (same as test data) means that
+        # unicode has been handled in same way, e.g. Pixel size symbols.
+        o2 = json.loads(json.dumps(o2))
+        assert o1 == o2
+
+
+class TestImages(IWebTest):
+    """Tests querying & editing Images."""
+
+    @pytest.fixture()
+    def user1(self):
+        """Return a new user in a read-annotate group."""
+        group = self.new_group(perms='rwra--')
+        user = self.new_client_and_user(group=group)
+        return user
+
+    @pytest.fixture()
+    def dataset_images(self, user1):
+        """Return Dataset with Images and an orphaned Image."""
+        query = get_query_service(user1)
+        dataset = DatasetI()
+        dataset.name = rstring('Dataset')
+
+        # Create 5 Images in Dataset
+        for i in range(5):
+            img = self.create_test_image(size_x=125, size_y=125,
+                                         session=user1[0].getSession(),
+                                         name="Image%s" % i)
+            img = ImageI(img.id.val, False)
+            dataset.linkImage(img)
+
+        # Import a single orphaned Image "tinyTest.d3d.dv" and get ImageID
```

Comment:
Could you avoid using ``tinyTest``, we want to get rid of it!
We can add methods to the testing framework to generate "richer" images.